src/concrete/ml/sklearn/base.py

```diff
@@ -1307,18 +1307,31 @@ def fit(self, X: Data, y: Target, **fit_parameters):
         # Check that the underlying sklearn model has been set and fit
         assert self.sklearn_model is not None, self._sklearn_model_is_not_fitted_error_message()
 
+        self._is_fitted = True
+
+        # Convert the tree inference with Numpy operators
+        # Adjust the auto rounder manually
+        self._convert_tree_to_numpy_and_compute_lsbs_to_remove(q_X, use_rounding=True)
+
+        return self
+
+    def _convert_tree_to_numpy_and_compute_lsbs_to_remove(
+        self, q_X: numpy.ndarray, use_rounding: bool
+    ):
+
+        assert self._is_fitted is True, "Model must be fitted"
+
+        check_array_and_assert(q_X)
+
         # Convert the tree inference with Numpy operators
         self._tree_inference, self.output_quantizers, self.onnx_model_ = tree_to_numpy(
-            self.sklearn_model,
-            q_X[:1],
+            self.sklearn_model,  # type: ignore[arg-type]
```

Comment:
why this type ignore ?

Reply:
are you sure ? I don't see any changes related to this and we are checking that `assert self.sklearn_model is not None` right above 🤔

Reply:
I really got a warning from mypy, otherwise I wouldn't have added it.

Now it's fixed